src/rgw/rgw_lc.cc

```diff
@@ -43,6 +55,49 @@ void RGWLifecycleConfiguration::_add_rule(LCRule *rule)
   prefix_map[rule->get_prefix()] = rule->get_expiration().get_days();
 }
 
+bool RGWLifecycleConfiguration::check_and_add_rule(LCRule *rule)
```

Comment:
Since add_rule() is unused now, could you just put this code in add_rule()?  That way, no one will accidentally call the wrong one.